src/components/Table/Table.Body.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react'
+
+import * as equal from 'fast-deep-equal'
+
+import Row from './Table.Row'
+
+import { BodyProps, BodyState } from './Table.types'
+
+export default class Body extends React.Component<BodyProps, BodyState> {
+  columnsCache = []
+  state: BodyState = { rows: [] }
+
+  constructor(props) {
+    super(props)
+    this.columnsCache = this.getColumnsCache(this.props.columns)
+
+    this.state.rows = this.getRows(props)
+  }
+
+  getRows(props: any = {}) {
+    const { isTableCollapsed, rows, maxRowsToDisplay } = props
+
+    if (!rows) {
+      return []
+    }
+
+    return isTableCollapsed ? rows.slice(0, maxRowsToDisplay) : [...rows]
+  }
+
+  getColumnsCache(columns) {
+    return columns.map(c => {
+      let key = c.columnKey
+      if (Array.isArray(key)) {
+        key = key.join(',')
+      }
+      return key
+    })
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
Is there a way to avoid using `componentWillReceiveProps` it's use is discouraged and will not work in future versions http://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops

Reply:
@tinkertrain I'm aware of the deprecating willReceiveProps as well. 🤔 

Thanks for the reminder! I'll be more aware the next time I feel like I need to use it :)

Reply:
TIL I learned about getDerivedStateFromProps :)

I updated the component to use that lifecycle method in cfa50283

Reply:
It's a React 16 thing :). So.. up until pretty recently, we wouldn't have been able to use it, heh